### PR TITLE
fix: throw error if no auth providers found

### PR DESCRIPTION
**Pull request by RobChangCA**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces error handling for authentication issues by adding a new error class and modifying the `AlchemySignerWebClient` to throw this error when no OAuth providers are found.

### Detailed summary
- Added `OAuthProvidersError` class extending `BaseError`.
- Updated `NotAuthenticatedError` class to override the `name` property.
- Modified `AlchemySignerWebClient` to throw `OAuthProvidersError` if `authProviders` is not found.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/signer/src/client/index.ts [hunk 1]
```diff
@@ -518,6 +518,11 @@ export class AlchemySignerWebClient extends BaseSignerClient<ExportWalletParams>
     } = args;
     const { codeChallenge, requestKey, authProviders } =
       await this.getOauthConfigForMode(mode);
+    if (!authProviders) {
+      throw new Error(
+        "No auth providers found, please enable auth providers on the alchemy dashboard here: https://dashboard.alchemy.com/accounts"
+      );
```
Comment (moldy530) [inline account-kit/signer/src/client/index.ts:524]: nit: can we add this to `errors.ts` as a new error type? you can set the docsPath to be `/react/getting-started`

File: account-kit/signer/src/client/index.ts [hunk 2]
```diff
@@ -670,8 +675,10 @@ export class AlchemySignerWebClient extends BaseSignerClient<ExportWalletParams>
     mode: OauthMode
   ): Promise<OauthConfig> => {
     if (this.oauthConfig) {
+      console.log("oauthConfig", this.oauthConfig);
```
Comment (moldy530) [inline account-kit/signer/src/client/index.ts:678]:
```suggestion
```

File: account-kit/signer/src/client/index.ts [hunk 3]
```diff
@@ -670,8 +675,10 @@ export class AlchemySignerWebClient extends BaseSignerClient<ExportWalletParams>
...
       return this.oauthConfig;
     } else if (mode === "redirect") {
+      console.log("initOauth");
```
Comment (moldy530) [inline account-kit/signer/src/client/index.ts:681]:
```suggestion
```

File: account-kit/signer/src/errors.ts [hunk 4]
```diff
@@ -13,3 +13,9 @@ export class NotAuthenticatedError extends BaseError {
     );
   }
 }
+
+export class OAuthProvidersError extends BaseError {
```
Comment (dphilipson) [inline account-kit/signer/src/errors.ts:17]: I see a lot of the other errors in this project having `override name = "ErrorName"`. Would be good to have that here I think (it looks like an oversight that it was missing from the other error in this file).

**Review by dphilipson — DISMISSED** [review]
Thanks for doing this! One suggestion, but should be good to go.

File: account-kit/signer/src/errors.ts (hunk 4 above)
Comment (RobChangCA) [reply account-kit/signer/src/errors.ts:17]: Sounds good, added to both errors in this file.

**Review by dphilipson — APPROVED** [review]
Thanks for the update, looks great!

**Pull request merged**